jersey-servers/src/main/java/com/palantir/remoting2/servers/jersey/CompressionFilter.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting2.servers.jersey;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.HttpHeaders;
+import com.jcraft.jzlib.DeflaterOutputStream;
+import com.jcraft.jzlib.GZIPOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.function.Function;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * Compresses responses based on the Accept-Encoding header in the request.
+ * <p>
+ * This is an alternative compression filter implementation that uses jzlib for compression rather than the java
+ * standard library due to performance issues observed on high core machines:
+ * https://bugs.openjdk.java.net/browse/JDK-8179001
+ */
+public final class CompressionFilter implements ContainerResponseFilter {
+
+    /**
+     * Supported encodings. Prefer gzip over deflate as Jetty does.
```

Comment:
referring to: http://www.eclipse.org/jetty/documentation/9.1.5.v20140505/gzip-filter.html

`If both gzip and deflate are specified in the accept-encoding header, then gzip will be used.`

can remove Jetty reference if you don't think it's helpful